feat: statistic card

kelvin3127 opened the pull request:
Statistic Card

### Description
A Statistic Card component
Includes theme styling
More can be added in future commits
Includes Storybook

--- Icons will be coming from icons component, temporarily just text.

### Testing Instructions

Open terminal and run "npm run dev"
Open application in browser with local host.
Import statistics card component and add to the template.
Statistics card component Props include:
theme - Changes the theme for the dropdown.
cardType - Changes the type of Statistic Card you would like, currently have only 1
data - The data being passed in
If no theme are is selected component will have it's own default theme.

### Testing Instructions Storybook

Open terminal and run "npm run storybook"
Open storybook in browser "http://localhost:6006/".
Select theme you would like to view

### Screenshots
Three Components
![1](https://github.com/LaunchUpCorp/Dashboard-Builder/assets/46061373/2b8fc6d8-ce2a-4590-824a-b4e6bd56a866)

Storybook
Three Themes
![2](https://github.com/LaunchUpCorp/Dashboard-Builder/assets/46061373/7ef6cd8b-18db-41e5-be38-2020715fae76)
![3](https://github.com/LaunchUpCorp/Dashboard-Builder/assets/46061373/5507a092-21d0-49df-b316-11e3d0da119d)
![4](https://github.com/LaunchUpCorp/Dashboard-Builder/assets/46061373/e21b1c21-7dd8-447f-b5dc-6d19f65c2fc7)

kelvin3127 committed:
fix: changes

did some changes

kelvin3127 committed:
feat: Statistics Card Component

Made a statistice Card Component
-Added Storybook
-Added Themes

kelvin3127 committed:
fix: remove app.vue code

Changed App.vue to defualt

kelvin3127 committed:
fix: formating

did some formatting and linting

YoussefLaunchUp commented on frontend/src/components/StatisticCard/StatisticCard.vue:
rename this to index.vue

YoussefLaunchUp commented on frontend/src/components/StatisticCard/StatisticCard.vue:
```diff
@@ -0,0 +1,110 @@
+    <div>
+        <div class="flex flex-col w-1/2 rounded-md border drop-shadow-md border-black" :class="themeType">
+            <div class="flex flex-row">
+                <p class="flex-grow mx-4 my-2">{{ cardType }} Icon</p> 
```
Although you are not using the icons right now, this component should be able to accept any icon independent from the type.

kelvin3127 replied:
Changes made, renamed Component to index.vue and added new prop for icon type to allow user choice of icon

kelvin3127 committed:
fix: Updates

Change name to index vue
Added new Property for icon type to allow user choice of icon

neli-rashkova reviewed (approved):
Good work!

YoussefLaunchUp reviewed (approved):
Great work

kelvin3127 committed:
Merge branch 'dev' into ckelvin3127/das-1-statistic-card
